### DISCUSSION TITLE
LVM to extend NSM folder across multiple disks

### DISCUSSION THREAD

**Discussion by cerberus10748** [Security-Onion-Solutions/securityonion#3508] [Unsupported Versions] [answered]
Hello , 
I have a sensor that i need massive storage on than one physical disk can provider. How would i expand the first hard disk to span across the second hard disk as one volume for the NSM folder? this would be after i already gave security onion up and running.

**Comment by albiondigitaladvisors** [comment, marked as answer]
I've used the following technique on one of my SO boxes.  Essentially you need to add a new physical drive to the LVM and then expand everything.  In my case, I added a second drive \dev\sdb, built an empty partition on that drive, and allocated the whole thing to LVM.  Then you extend the Volume Group, and finally extend the Logical Volume.

Initializing disk for LVM

df -hm			#  check current allocations

sudo fdisk /dev/sdb
o<enter>	#  create empty partition on disk
n<enter>	#  create new partition
	#  keep pressing <enter> and take all defaults
t<enter>	#  set partition type	
8e<enter>	#    to Linux LVM
w<enter>	#  write changes to disk

Adding the disk to LVM PV

sudo pvcreate /dev/sdb1			#  add disk partition to LVM as PV (Note the VG name)
sudo pvscan					#  list all the PVs
sudo pvdisplay /dev/sdb1		#  list detail about a specific PV

Extend Volume Group (VG)

sudo vgextend <VG name> /dev/sdb1	#  VG from above. /dev/sdb1 is the newly allocated partition.

Extend Logical Volume (LV)

sudo lvextend -r /dev/<VG name>/nsm /dev/sdb1
	#  where VG name is from above, and /nsm is the filesystem within it, to extend.
	#  Entire /dev/sdb1 volume will be consumed.
        #  Note the -r is important; it extends /nsm into the LV

	#  sudo pvs will help find the VG name	
	#  sudo lvs will help find the logical volume (LV) name 

df -hm		#  check allocations
pvscan		#  check allocations

Good luck.

**Reply by cerberus10748** [reply]
You mention VGname in the last step in extend logical. Where is that in that command that i enter my name?

**Reply by albiondigitaladvisors** [reply]
Sorry, the comment above is on the wrong line.  If you issue the command 

sudo pvscan
it will list the existing Physical Volumes, and the Volume Groups that they belong to, like:

$  sudo pvscan
   PV /dev/sda3   VG system          lvm2 [<931.02 GiB / 0    free]
 the VG name is the word immediately following the text "VG" in the output line for the Physical Volume of interest.  In this case  /dev/sda3 is the Physical Volume, and "system" is the Volume Group.

Eratta:  also please note a typo in the lvextend command as shown above.  The correct command should be:
    sudo lvextend -r /dev/nsm /dev/sdb1
with a single / between dev and nsm.

**Reply by cerberus10748** [reply]
This worked perfectly. So I am assuming i would do the same with a different drive name to add a 3rd and vice versa.

**Reply by albiondigitaladvisors** [reply]
Glad it worked for you.  Yes, you are correct.  You can add more PVs to the VG by changing the name accordingly -- sdb would become sdc, sdd, etc.

**Comment by albiondigitaladvisors** [comment]
Answer given above has been accepted.
